lib/checkother.cpp

```diff
@@ -448,8 +448,9 @@ void CheckOther::checkRedundantAssignment()
                         continue;
                 }
 
+                const Token* rhs = tok->astOperand2();
                 // Do not warn about assignment with 0 / NULL
-                if (Token::simpleMatch(tok->astOperand2(), "0") || FwdAnalysis::isNullOperand(tok->astOperand2()))
+                if ((rhs && rhs->isNumber() && MathLib::isNullValue(rhs->str())) || isNullOperand(rhs))
```

Comment:
Is this `rhs->isNumber()` needed?

Reply:
I'll remove it.

Reply:
Actually, we need it because `isNullValue` doesn't check that it's input is a number, causing for example `isNullValue(-ENOMEM)` to be true... I'll remove the check here and add the check to `isNullValue()` instead.